src/main/resources/api/service_openapi.yaml

```diff
@@ -598,23 +594,20 @@ components:
 
     CreateDataReferenceRequestBody:
       type: object
-      required: [name, cloningInstructions]
+      required: [name, referenceType, cloningInstructions]
```

Comment:
I think that's a good point. default it to data repo snapshot and leave it optional.

Reply:
Rawls is already specifying type, and this was previously enforced by rejecting requests with no `referenceType` field in `ControllerValidationUtils`. 
We did that validation as part of the app logic instead of putting it in Swagger because the request shape was (kind of) also used for controlled resource requests, so only common fields between the two were made required. Now that we're removing all the unused controlled resource stuff here, we can move `referenceType` to required. I'd also like to move `referenceInfo` there once we stop using the `reference` field.